### PR TITLE
Add devcontainer

### PR DESCRIPTION
**Pull request by NickBorgersOnLowSecurityNode**
This PR adds support for [Devcontainers](https://containers.dev/), a thing coming out of Microsoft which they've now made a more open spec.

I was exploring using glauth for a usecase at Pinterest, and did this as an initial step to make it easier to start working with it. It now looks like what I'm trying to accomplish wouldn't be most easily done on top of `glauth` because the usecase is quite different, but sharing this in case you want it.

Today the only two IDEs I know that support Devcontainers are VS Code and GitHub. I'm generally very pro-containerized development environments as it makes it easier for developers to get consistent experiences and therefore share code for helping each other fix things; this Devcontainers thing is pretty much what I was looking for.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			"label": "LDAP",
+			"onAutoForward": "notify"
+		},
+		"5555": {
+			"label": "Web UI",
+			"onAutoForward": "openBrowser"
+		}
+	},
+
+	"postCreateCommand": "cd v2 && make setup"
```
Comment (NickBorgersOnLowSecurityNode) [inline .devcontainer/devcontainer.json:53]: Still uses the existing `setup` command in the `Makefile`, just wrapped in a container

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"image": "mcr.microsoft.com/devcontainers/go:0-1.19-bullseye",
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	"features": {
+		"docker-in-docker": {
+			"version": "latest",
+			"moby": true
+		},
+		"ghcr.io/rocker-org/devcontainer-features/apt-packages:1": {
+			"packages": "oathtool,ldap-utils"
```
Comment (NickBorgersOnLowSecurityNode) [inline .devcontainer/devcontainer.json:15]: Installs the packages into the Devcontainer for development

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/go
+{
+	"name": "glauth",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/go:0-1.19-bullseye",
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	"features": {
+		"docker-in-docker": {
```
Comment (NickBorgersOnLowSecurityNode) [inline .devcontainer/devcontainer.json:10]: A docker instance is available inside the Devcontainer for the containerized tests of `make testdocker`

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+				"ms-azuretools.vscode-docker"
+			]
+		}
+	},
+
+	"containerEnv": {
+		"REPO": "glauth"
+	},
+
+	"capAdd": ["SYS_PTRACE"],
```
Comment (NickBorgersOnLowSecurityNode) [inline .devcontainer/devcontainer.json:35]: Need this permission for Golang debugging

**Comment by Fusion** [comment]
Reviewing!

File: go.work [hunk 5]
```diff
@@ -0,0 +1,3 @@
+go 1.16
```
Comment (Fusion) [inline go.work:1]: A minor nit: `go work` was added in go 1.18

**Comment by Fusion** [comment]
Merging this now, after updating to Go1.19+

**Pull request merged**